src/explorer/Components/SampleValuesGeneratorConfig.cs

```diff
@@ -0,0 +1,63 @@
+namespace Explorer.Components
+{
+    using System.Threading.Tasks;
+
+    using Diffix;
+
+    public class SampleValuesGeneratorConfig : ExplorerComponent<SampleValuesGeneratorConfig.Result>
+    {
+        public const int DefaultNumValuesToPublish = 20;
+        public const int DefaultDistinctValuesBySamplesToPublishRatioThreshold = 5;
+
+        private readonly ResultProvider<DistinctValuesComponent.Result> distinctValuesProvider;
+
+        public SampleValuesGeneratorConfig(ResultProvider<DistinctValuesComponent.Result> distinctValuesProvider)
+        {
+            this.distinctValuesProvider = distinctValuesProvider;
+        }
+
+        public int NumValuesToPublish { get; set; } = DefaultNumValuesToPublish;
+
+        public int DistinctValuesBySamplesToPublishRatioThreshold { get; set; } = DefaultDistinctValuesBySamplesToPublishRatioThreshold;
+
+        protected override async Task<Result?> Explore()
+        {
+            var distinctValuesResult = await distinctValuesProvider.ResultAsync;
+            if (distinctValuesResult == null)
+            {
+                return null;
+            }
+
+            var valueCounts = distinctValuesResult.ValueCounts;
+            var categoricalSampling = valueCounts.IsCategorical;
+            var minRowsForCategoricalSampling = DistinctValuesBySamplesToPublishRatioThreshold * NumValuesToPublish;
```

Comment:
I'm not sure we should scale the minimum required rows according to the number of values to be published - what happens if we want to generate 1000 values instead of 20? This shouldn't change the result, but it might mean that generating 20 vs 1000 values for the same dataset gives totally different text samples.  

Maybe we can use something much simpler, like use categorical sampling as long as the distinct values make up ~half of the original dataset, for example?

Reply:
I vote for a heuristic of:
- if there are known real values, then use those
- if there are _no_ real values, then synthesize

The known real values are ~99% of the cases bound to be much better than the synthesized ones.

Reply:
Yes, in the case of text columns the real values are almost always better than the synthesized ones.
With this condition I wanted to avoid some extreme cases, like when just a very small number of values are unsuppressed and we want to generate a large number of sample values. For example for the `loans` table in `gda_banking` dataset we have just 10 unsuppressed distinct values for the `lastname` (from around 800 total rows in the table). So I thought that it will somehow be misleading to generate 20 sample values (the default count for generated sample values) from just 10 distinct values.

Reply:
The current condition for performing categorical sampling for text column is this:
```
            var minRowsForCategoricalSampling = DistinctValuesBySamplesToPublishRatioThreshold * NumValuesToPublish;
            if (Context.ColumnInfo.Type == DValueType.Text && valueCounts.NonSuppressedRows > minRowsForCategoricalSampling)
            {
                categoricalSampling = true;
            }
```

@sebastian, do you suggest to change it to something like this?
```
            var minRowsForCategoricalSampling = 5;
            if (Context.ColumnInfo.Type == DValueType.Text && valueCounts.NonSuppressedRows > minRowsForCategoricalSampling)
            {
                categoricalSampling = true;
            }
```

Reply:
Yes, that would probably be better. But what about the case where the text column is in fact categorical. Say it only contains the values `male`, `female`, `other`. In that case there are likely no unknown or suppressed values, but also it doesn't pass your limit check.

Maybe the check should rather be that if the amount of suppressed rows is say more than 95% or something like that, then we synthesize. Otherwise we sample from the known values?